worldedit-core/src/main/java/com/boydti/fawe/beta/implementation/lighting/HeightMapType.java

```diff
@@ -1,5 +1,46 @@
 package com.boydti.fawe.beta.implementation.lighting;
 
+import com.sk89q.worldedit.registry.state.Property;
+import com.sk89q.worldedit.registry.state.PropertyKey;
+import com.sk89q.worldedit.world.block.BlockCategories;
+import com.sk89q.worldedit.world.block.BlockState;
+
 public enum HeightMapType {
-    MOTION_BLOCKING, MOTION_BLOCKING_NO_LEAVES, OCEAN_FLOOR, WORLD_SURFACE
+    MOTION_BLOCKING {
+        @Override
+        public boolean blocks(BlockState state) {
+            return state.getMaterial().isSolid() || HeightMapType.isFluid(state);
+        }
+    },
+    MOTION_BLOCKING_NO_LEAVES {
+        @Override
+        public boolean blocks(BlockState state) {
+            return (state.getMaterial().isSolid() || HeightMapType.isFluid(state)) && !HeightMapType.isLeaf(state);
+        }
+    },
+    OCEAN_FLOOR {
+        @Override
+        public boolean blocks(BlockState state) {
+            return state.getMaterial().isSolid();
+        }
+    },
+    WORLD_SURFACE {
+        @Override
+        public boolean blocks(BlockState state) {
```

Comment:
"blocks" is a little ambiguous given Minecraft ***blocks***. Might be better to be "checkBlocksLight" or "blocksLight".

Reply:
Not about light here :p that's why I didn't came up with something better

Reply:
Something like "includes" might be preferrable then. It would thus be `HeightMapType.<TYPE>.includes`. I also added the comment to the bottom one, rather than any heightmap type in particular aha